### ISSUE TITLE
Support Wing in GitHub syntax highlighting (linguist)

### ISSUE THREAD

**Issue by eladb** [winglang/wing#1038]
### Community Note

> Please vote by adding a 👍 reaction to the issue to help us prioritize.
> If you are interested to work on this issue, please leave a comment.

### Feature Spec

> NOTE: This is likely not something we can do right now because GitHub are requiring "in the wild" usage of Wing backed by code search results.

GitHub now recognizes Wing files with syntax highlighting:

1. Any `.w` file
2. Markdown code blocks: ```w

Any other implications (code search etc).

### Use Cases

Obvious

### Implementation Notes

See [adding-a-language](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language) in the contribution guide for [linguist](https://github.com/github/linguist).

### Component

Other

**Comment by eladb** [comment]
Keep

**Comment by MarkMcCulloh** [comment]
The current guideline for when they would consider adding a new language is [here](https://github.com/github-linguist/linguist/issues/5756):

> - at least 2000 files per extension indexed in the last year (the number you see at the top of the search results), unless the extension is expected to only occur once per repo, then 200 files.
> - with a reasonable distribution across unique :user/:repo combinations assessed by manually and randomly clicking through the results.

> If particular users are showing a high proportion of the results, I'll manually filter out those users using -user:<username> to reduce their impact on my assessment.

Given this, we're still a while away from being considered.

**Comment by MarkMcCulloh** [comment]
The usual solution would highlighting through a [remark](https://github.com/remarkjs/remark) plugin. Ideally, we could use something we already have for highlighting (either tree-sitter grammar+query or textmate grammar). 
I haven't tried it, but [shiki](https://shiki.matsu.io/) seems like a good choice.

[plugin](https://github.com/stefanprobst/remark-shiki) + [here's how to load our custom grammar](https://github.com/stefanprobst/remark-shiki#supported-languages)

**Comment by Chriscbr** [comment]
> I haven't tried it, but [shiki](https://shiki.matsu.io/) seems like a good choice.

[shiki-twoslash](https://shikijs.github.io/twoslash/) also looks pretty cool

**Comment by MarkMcCulloh** [comment]
> > I haven't tried it, but [shiki](https://shiki.matsu.io/) seems like a good choice.
> 
> [shiki-twoslash](https://shikijs.github.io/twoslash/) also looks pretty cool

Yep it's super cool, although unfortunately not reasonably usable for wing. [Maybe one day though](https://github.com/shikijs/twoslash/issues/149)

**Comment by eladb** [comment]
Created https://github.com/github-linguist/linguist/issues/6585

**Comment by eladb** [comment]
I think we qualify for adding Wing to GitHub ([query](https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.w+%22bring+cloud%22)).

**Comment by Chriscbr** [comment]
To anyone looking for an alternative, you can download this Chrome extension that adds syntax highlighting when you're browsing GitHub: https://chromewebstore.google.com/detail/winglang-syntax-hightligh/gjnleleianfjpmckmmdeahlklhcdlakj
